### DISCUSSION TITLE
关于编译后打包成Debian系发行版的deb安装包的一些探索

### DISCUSSION THREAD

**Discussion by ZeaKyX** [wh201906/SerialTest#83] [Ideas]
如题，算是对 https://github.com/wh201906/SerialTest/discussions/40 尝试填坑
主要是无意间发现了一个感觉很好用的懒人神器——[makedeb](https://www.makedeb.org/)

> makedeb 是一款适用于 Debian 和 Ubuntu 系统的打包工具，其目标是简单易用，同时与标准 Debian 工具一样功能强大。

这样就可以根据本包的[AUR源码](https://aur.archlinux.org/packages/serialtest)直接打包成deb包，当然需要一些小修改，主要就是依赖包的名字（毕竟各个distro的包名不尽相同）
我所做的修改如下：
```
diff --git a/PKGBUILD b/PKGBUILD
index b3a0303..d5bd72a 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -12,8 +12,8 @@ license=('GPL-3.0-only')
 provides=(${_pkgname})
 conflicts=(${pkgname}-git)
 #replaces=(${pkgname})
-depends=('qcustomplot' 'qt5-serialport' 'qt5-connectivity')
-makedepends=('qt5-tools')
+depends=('libqcustomplot-dev' 'libqt5serialport5-dev' 'qtconnectivity5-dev')
+makedepends=('qt5-qmake')
 backup=()
 options=('!strip')
 install=${pkgname}.install
```
基于ARM64 & Ubuntu20.04 简单试了下使用没毛病，这边主要担心的就是`libqcustomplot-dev`这个包在各个系统下的版本不同，我用的是2.0.1的:
![image](https://github.com/user-attachments/assets/ae892233-52f2-499f-8354-a850000c9f1b)
但是本项目默认编译是下载的prebuilt的2.1.1版本，@wh201906 需要大佬帮忙看下是否支持这么大的版本跨度？

**Comment by wh201906** [comment]
我之后可能会考虑把QCustomPlot整个库以源代码的形式放到项目里面，这样方便编译一些。

**Reply by ZeaKyX** [reply]
好的，只是这样似乎会臃肿些
我在想尝试写一下这个直接生成deb包的action了，这样后期也可以自动release，这个AUR的包是您维护吗，qcustomplot加入后那后面这个包流程估计也要改？

**Reply by wh201906** [reply]
AUR的包不是我在管理，但是我有权限修改里面的内容。后期等加入QCustomPlot之后PKGBUILD应该是要改的。
